lib/src/message/trailer.rs

```diff
@@ -65,31 +84,24 @@ impl<'a> Iterator for Trailers<'a> {
     type Item = Trailer;
 
     fn next(&mut self) -> Option<Self::Item> {
-        lazy_static! {
-            static ref RE: Regex = Regex::new(r"^(?P<key>(.*)):\ (?P<value>(.*))$").unwrap();
-        }
+        let mut at_end = false;
 
         loop {
-            match self.0.next() {
-                None => return None,
-                Some(line) => {
-                    match RE.captures(line) {
-                        None => continue,
-                        Some(capture) => {
-                            let key = capture.name("key").map(|m| {
-                                TrailerKey(String::from(m.as_str()))
-                            }).unwrap(); // TODO: fix unwrap()
-
-                            let value = capture.name("value").map(|m| {
-                                TrailerValue::from_slice(m.as_str())
-                            }).unwrap(); // TODO: fix unwrap()
-
-                            return Some(Trailer {
-                                key  : key,
-                                value: value,
-                            });
-                        }
-                    }
+            if let Some(trailer) = self.buf.pop_front() {
+                return Some(trailer);
+            }
+
+            if at_end {
+                return None
+            }
+
+            // refill buffer from next block
+            loop {
```

Comment:
I would use [loop labels](https://doc.rust-lang.org/book/loops.html#loop-labels) so one can easily see which loop a `break` is for.